data-jdbc/src/test/groovy/io/micronaut/data/jdbc/oraclexe/jsonview/StudentViewRepository.java

```diff
@@ -0,0 +1,22 @@
+package io.micronaut.data.jdbc.oraclexe.jsonview;
+
+import io.micronaut.data.annotation.Query;
+import io.micronaut.data.annotation.TypeDef;
+import io.micronaut.data.jdbc.annotation.JdbcRepository;
+import io.micronaut.data.model.DataType;
+import io.micronaut.data.model.query.builder.sql.Dialect;
+import io.micronaut.data.repository.PageableRepository;
+
+import java.util.Optional;
+
+@JdbcRepository(dialect = Dialect.ORACLE)
+public interface StudentViewRepository extends PageableRepository<StudentView, Long> {
```

Comment:
The alternative syntax might be:
`public interface StudentViewRepository extends PageableRepository<@JsonView StudentView, Long>`
or
`public interface StudentViewRepository extends PageableRepository<@QueryResult(JSON) StudentView, Long>``
WDYT @graemerocher ?

Reply:
makes sense to me

Reply:
the first one looks cleaner but I am fine with either

Reply:
I don't know what is wrong with having `JsonView` on the entity? It is now defined as an alias for `EntityRepresentation` with JSON column.
It will be needed anyway later when we will add support for schema creation, need to be able to distinguish regular `MappedEntity` that creates table from the entity that will produce json view. And I also need it [here](https://github.com/micronaut-projects/micronaut-data/pull/2163/files#diff-98283ce89d0eae1bcc418c6b79cc563775312f793acd4d02eb7fae3404600cde)